orderable/models.py

```diff
@@ -67,6 +67,20 @@ def prev(self):
             sort_order__lt=self.sort_order
         ).order_by('-sort_order').first()
 
+    @staticmethod
+    def _update(qs):
+        """
+        Increment the sort_order in a queryset.
+
+        Handle IntegrityErrors caused by unique constraints.
+        """
+        try:
+            with transaction.atomic():
+                qs.update(sort_order=models.F('sort_order') + 1)
+        except IntegrityError:
+            for obj in qs.order_by('-sort_order'):
+                qs.filter(pk=obj.pk).update(sort_order=models.F('sort_order') + 1)
```

Comment:
Sounds like a manifestation of https://github.com/incuna/django-orderable/issues/7